packages/contracts-bedrock/src/cannon/libraries/MIPSInstructions.sol

```diff
@@ -1,12 +1,178 @@
 // SPDX-License-Identifier: MIT
 pragma solidity 0.8.15;
 
+import { MIPSMemory } from "src/cannon/libraries/MIPSMemory.sol";
 import { MIPSState as st } from "src/cannon/libraries/MIPSState.sol";
 
 library MIPSInstructions {
+    /// @param _pc The program counter.
+    /// @param _memRoot The current memory root.
+    /// @param _insnProofOffset The calldata offset of the memory proof for the current instruction.
+    /// @return insn_ The current 32-bit instruction at the pc.
+    /// @return opcode_ The opcode value parsed from insn_.
+    /// @return fun_ The function value parsed from insn_.
+    function getInstructionDetails(
+        uint32 _pc,
+        bytes32 _memRoot,
+        uint256 _insnProofOffset
+    )
+        internal
+        pure
+        returns (uint32 insn_, uint32 opcode_, uint32 fun_)
+    {
```

Comment:
To be safe, let's wrap this function body `unchecked` so that we're explicit that we want the same under/overflow behavior as the Go VM.

We should also wrap the other every library functions in `unchecked`.

Reply:
Great call - done